wp-seo-main.php

```diff
@@ -263,6 +263,18 @@ function wpseo_on_activate_blog( $blog_id ) {
 	}
 }
 
+/**
+ * Alternative method for calling wpseo_on_activate_blog, for when supplied with a WP_site object instead of an ID.
+ *
+ * @param object $blog The WP_Site object received from wp_insert_site.
+ *
+ * @return void
+ */
+function wpseo_on_activate_blog_from_wp_site( $blog ) {
+	if ( is_object( $blog ) && isset( $blog->blog_id ) ) {
+		wpseo_on_activate_blog( (int) $blog->blog_id );
```

Comment:
Doublechecking: is it intentional that nothing happens when there is no blog_id found?

Reply:
We need the blog id, when it's not found the wrong argument has been passed to the hook.